dash/include/dash/halo/Halo.h

```diff
@@ -281,137 +277,141 @@ std::ostream& operator<<( std::ostream&                        os,
 
 template <dim_t NumDimensions>
 class HaloSpec {
- public:
-  using self            = HaloSpec<NumDimensions>;
-  using HaloRegionSpecT = HaloRegionSpec<NumDimensions>;
-  using region_index_t  = typename HaloRegionSpecT::region_index_t;
-  using extent_t        = typename HaloRegionSpecT::extent_t;
-  using SpecsT   = std::array<HaloRegionSpecT, HaloRegionSpecT::MaxIndex>;
-  using StencilT = Stencil<NumDimensions>;
-
- public:
-  constexpr HaloSpec( const SpecsT& specs ) : _specs( specs ) {}
+private:
+  using Self_t          = HaloSpec<NumDimensions>;
+  using RegionCoords_t  = RegionCoords<NumDimensions>;
+  using RegionSpec_t    = RegionSpec<NumDimensions>;
+  using Specs_t         = std::array<RegionSpec_t, RegionCoords_t::MaxIndex>;
+  using Stencil_t       = Stencil<NumDimensions>;
+  using region_index_t  = typename RegionCoords_t::region_index_t;
+  using region_size_t   = typename RegionCoords_t::region_index_t;
+  using region_extent_t = typename RegionSpec_t::region_extent_t;
```

Comment:
The types are declared private but exposed to the user through member functions. In that case, they should be public (I know this is messy as well all around DASH). Please also check other places for this.